src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -930,6 +930,15 @@
   <data name="DoNotUseObsoleteKDFAlgorithmMessage" xml:space="preserve">
     <value>Call to obsolete key derivation function {0}.{1}</value>
   </data>
+  <data name="DoNotUseReferenceEqualsWithValueTypesDescription" xml:space="preserve">
+    <value>Value type typed arguments are uniquely boxed for each call to this method, therefore the result is always false.</value>
+  </data>
+  <data name="DoNotUseReferenceEqualsWithValueTypesMessage" xml:space="preserve">
+    <value>The argument for the '{0}' parameter to ReferenceEquals evaluates to the value type '{1}'. Due to value boxing, this call to ReferenceEquals will always return false.</value>
```

Comment:
Wondering if the message should have the user actionable part at the beginning. Also wondering if the parameter name in the message is actually helpful? Maybe we can avoid it to simplify the message.

```suggestion
    <value>Do not pass an argument with value type '{0}' to 'ReferenceEquals'. Due to value boxing, this call to 'ReferenceEquals' will always return 'false'.</value>
```

I would be fine to even go a step further and simplify to `Do not pass a value type argument to 'ReferenceEquals'. Due to value boxing, this call to 'ReferenceEquals' will always return 'false'.`. I'll let you make a call on the preferred message.

Reply:
I feel like having the type name in the message is helpful, especially for non-IDE experiences.

The line and character positions do help solve the "which argument" problem, so I can see leaving that one out.

So I think I'm leaning toward your first suggestion ("Do not pass an argument with value type '{0}' to 'ReferenceEquals'. Due to value boxing, this call to 'ReferenceEquals' will always return 'false'.")